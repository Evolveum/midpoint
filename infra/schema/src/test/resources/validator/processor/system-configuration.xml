<!--
  ~ Copyright (C) 2010-2023 Evolveum and contributors
  ~
  ~ This work is dual-licensed under the Apache License 2.0
  ~ and European Union Public License. See LICENSE file for details.
  -->
<systemConfiguration xmlns="http://midpoint.evolveum.com/xml/ns/public/common/common-3"
        oid="3947d3b1-b6e8-46ec-b382-1bde67da6f07">

    <name>system-configuration</name>

    <notificationConfiguration>
        <handler>
            <simpleUserNotifier>
                <transport>custom1</transport>
            </simpleUserNotifier>
        </handler>
        <mail name="m2">
            <defaultFrom>example@example.com</defaultFrom>
            <server>
                <host>localhost</host>
            </server>
        </mail>
        <file name="f1">
            <file>a.csv</file>
            <redirectToFile>redirect.csv</redirectToFile>
            <debug>true</debug>
            <whiteList>asdf</whiteList>
            <blackList>jklo</blackList>
            <recipientFilterExpression>
                <script>
                    <code>
                        return null
                    </code>
                </script>
            </recipientFilterExpression>
        </file>
        <sms name="s1">
            <gateway>
                <urlExpression>
                    <script>
                        <code>
                            return 'one.example'
                        </code>
                    </script>
                </urlExpression>
            </gateway>
        </sms>
        <sms name="s2">
            <redirectToFile>redirected.log</redirectToFile>
            <blackList>blacklisted@example.com</blackList>
            <debug>true</debug>
            <logToFile>log.log</logToFile>
            <recipientFilterExpression>
                <script>
                    <code>
                        return null
                    </code>
                </script>
            </recipientFilterExpression>
            <whiteList>whitelisted@example.com</whiteList>
            <gateway>
                <method>get</method>
                <urlExpression>
                    <script>
                        <code>
                            return 'example.com'
                        </code>
                    </script>
                </urlExpression>
                <username>user</username>
                <password>pass</password>
            </gateway>
        </sms>
        <customTransport name="custom1">
            <redirectToFile>notification-redirected.log</redirectToFile>
            <expression>
                <script>
                    <code>
                        return null
                    </code>
                </script>
            </expression>
        </customTransport>
        <customTransport name="custom2">
            <redirectToFile>notification-redirected2.log</redirectToFile>
            <expression>
                <script>
                    <code>
                        return null
                    </code>
                </script>
            </expression>
        </customTransport>
    </notificationConfiguration>

    <roleManagement>
        <defaultAssignmentConstraints>
            <allowSameTarget>true</allowSameTarget>
            <allowSameRelation>true</allowSameRelation>
        </defaultAssignmentConstraints>
        <roleCatalogRef oid="0e5b7304-ea5c-438e-84d1-2b0ce40517ce" type="OrgType"/>
        <defaultCollection>
            <collectionUri>allRoles</collectionUri>
        </defaultCollection>
        <roleCatalogCollections>
            <collection>
                <collectionUri>allRoles</collectionUri>
            </collection>
            <collection>
                <collectionUri>myCustomCollection</collectionUri>
            </collection>
        </roleCatalogCollections>
    </roleManagement>

    <adminGuiConfiguration>
        <userDashboard>
            <widget id="14">
                <identifier>123</identifier>
                <visibility>hidden</visibility>
            </widget>
            <widget id="15">
                <identifier>456</identifier>
                <visibility>vacant</visibility>
            </widget>
        </userDashboard>
        <userDashboardLink id="12">
            <targetUrl>https://google.com</targetUrl>
            <color>red</color>
            <icon>
                <cssClass>black</cssClass>
            </icon>
        </userDashboardLink>
        <userDashboardLink id="13">
            <targetUrl>https://sample.com</targetUrl>
            <label>Sample</label>
        </userDashboardLink>
        <objectForms>
            <objectForm>
                <type>OrgType</type>
                <formSpecification>
                    <title>Transform</title>
                    <panelClass>com.example.midpoint.gui.forms.ConfigurationTableTabPanel</panelClass>
                </formSpecification>
            </objectForm>
        </objectForms>
        <objectCollectionViews>
            <objectCollectionView id="3">
                <action id="4">
                    <name>action1</name>
                </action>
                <action id="5">
                    <name>action2</name>
                </action>
                <searchBoxConfiguration>
                    <defaultScope>oneLevel</defaultScope>
                    <defaultObjectType>UserType</defaultObjectType>
                    <searchItems>
                        <searchItem id="7">
                            <displayName>sample1</displayName>
                        </searchItem>
                    </searchItems>
                </searchBoxConfiguration>
            </objectCollectionView>
            <objectCollectionView id="6">
                <searchBoxConfiguration>
                    <defaultScope>oneLevel</defaultScope>
                    <defaultObjectType>RoleType</defaultObjectType>
                    <scopeConfiguration>
                        <defaultValue>subtree</defaultValue>
                    </scopeConfiguration>
                    <objectTypeConfiguration>
                        <defaultValue>OrgType</defaultValue>
                    </objectTypeConfiguration>
                    <searchItems>
                        <searchItem id="8">
                            <displayName>sample2</displayName>
                            <display>
                                <label>sample3</label>
                            </display>
                        </searchItem>
                    </searchItems>
                </searchBoxConfiguration>
            </objectCollectionView>
            <objectCollectionView id="15">
                <identifier>orgMember</identifier>
                <type>OrgType</type>
                <additionalPanels>
                    <memberPanel>
                        <searchBoxConfiguration>
                            <scopeConfiguration>
                                <display>
                                    <label>Custom Scope</label>
                                    <help>Help scope text</help>
                                </display>
                                <defaultValue>subtree</defaultValue>
                            </scopeConfiguration>
                            <objectTypeConfiguration>
                                <display>
                                    <label>Custom Type</label>
                                </display>
                                <defaultValue>OrgType</defaultValue>
                                <supportedTypes>OrgType</supportedTypes>
                                <supportedTypes>UserType</supportedTypes>
                                <supportedTypes>ServiceType</supportedTypes>
                            </objectTypeConfiguration>
                            <indirectConfiguration>
                                <visibility>hidden</visibility>
                            </indirectConfiguration>
                        </searchBoxConfiguration>
                    </memberPanel>
                </additionalPanels>
            </objectCollectionView>
        </objectCollectionViews>
        <objectDetails>
            <objectDetailsPage id="10">
                <type>UserType</type>
                <forms id="16">
                    <type>UserType</type>
                    <formSpecification>
                        <title>Transform</title>
                        <panelClass>com.example.midpoint.gui.forms.ConfigurationTableTabPanel</panelClass>
                    </formSpecification>
                </forms>
                <container id="9">
                    <identifier>123</identifier>
                </container>
            </objectDetailsPage>
        </objectDetails>
    </adminGuiConfiguration>

    <defaultObjectPolicyConfiguration id="2">
        <propertyConstraint>
            <path>name</path>
            <visibility>visible</visibility>
        </propertyConstraint>
        <propertyConstraint>
            <path>givenName</path>
            <visibility>hidden</visibility>
        </propertyConstraint>
        <itemConstraint>
            <path>familyName</path>
            <visibility>visible</visibility>
        </itemConstraint>
    </defaultObjectPolicyConfiguration>
    <cleanupPolicy>
        <objectResults>
            <maxAge>P1D</maxAge>
            <maxRecords>10</maxRecords>
        </objectResults>
    </cleanupPolicy>
    <workflowConfiguration>
        <useLegacyApproversSpecification>always</useLegacyApproversSpecification>
    </workflowConfiguration>
</systemConfiguration>
