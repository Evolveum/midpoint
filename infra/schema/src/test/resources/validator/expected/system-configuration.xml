<!--
  ~ Copyright (C) 2010-2023 Evolveum and contributors
  ~
  ~ This work is dual-licensed under the Apache License 2.0
  ~ and European Union Public License. See LICENSE file for details.
  -->
<systemConfiguration xmlns="http://midpoint.evolveum.com/xml/ns/public/common/common-3"
        xmlns:c="http://midpoint.evolveum.com/xml/ns/public/common/common-3"
        oid="3947d3b1-b6e8-46ec-b382-1bde67da6f07">

    <name>system-configuration</name>


    <messageTransportConfiguration>
        <file>
            <name>f1</name>
            <file>a.csv</file>
            <redirectToFile>redirect.csv</redirectToFile>
            <debug>true</debug>
            <whiteList>asdf</whiteList>
            <blackList>jklo</blackList>
            <recipientFilterExpression>
                <script>
                    <code>
                        return null
                    </code>
                </script>
            </recipientFilterExpression>
        </file>
        <mail>
            <name>m2</name>
            <defaultFrom>example@example.com</defaultFrom>
            <server>
                <host>localhost</host>
            </server>
        </mail>
        <sms>
            <name>s1</name>
            <gateway>
                <urlExpression>
                    <script>
                        <code>
                            return 'one.example'
                        </code>
                    </script>
                </urlExpression>
            </gateway>
        </sms>
        <sms>
            <name>s2</name>
            <redirectToFile>redirected.log</redirectToFile>
            <blackList>blacklisted@example.com</blackList>
            <debug>true</debug>
            <logToFile>log.log</logToFile>
            <recipientFilterExpression>
                <script>
                    <code>
                        return null
                    </code>
                </script>
            </recipientFilterExpression>
            <whiteList>whitelisted@example.com</whiteList>
            <gateway>
                <method>get</method>
                <urlExpression>
                    <script>
                        <code>
                            return 'example.com'
                        </code>
                    </script>
                </urlExpression>
                <username>user</username>
                <password>pass</password>
            </gateway>
        </sms>
        <customTransport>
            <name>custom1</name>
            <redirectToFile>notification-redirected.log</redirectToFile>
            <expression>
                <script>
                    <code>
                        return null
                    </code>
                </script>
            </expression>
        </customTransport>
        <customTransport>
            <name>custom2</name>
            <redirectToFile>notification-redirected2.log</redirectToFile>
            <expression>
                <script>
                    <code>
                        return null
                    </code>
                </script>
            </expression>
        </customTransport>
    </messageTransportConfiguration>

    <notificationConfiguration>
        <handler>
            <simpleUserNotifier>
                <transport>custom1</transport>
            </simpleUserNotifier>
        </handler>
    </notificationConfiguration>

    <roleManagement>
        <defaultAssignmentConstraints>
            <allowSameTarget>true</allowSameTarget>
            <allowSameRelation>true</allowSameRelation>
        </defaultAssignmentConstraints>
    </roleManagement>

    <adminGuiConfiguration>
        <homePage>
            <identifier>user</identifier>
            <type>c:UserType</type>
            <widget>
                <display>
                    <icon>
                        <cssClass>black red</cssClass>
                    </icon>
                </display>
                <panelType>linkWidget</panelType>
                <action>
                    <identifier>link</identifier>
                    <target>
                        <targetUrl>https://google.com</targetUrl>
                    </target>
                </action>
            </widget>
            <widget>
                <display>
                    <label>Sample</label>
                </display>
                <panelType>linkWidget</panelType>
                <action>
                    <identifier>link</identifier>
                    <target>
                        <targetUrl>https://sample.com</targetUrl>
                    </target>
                </action>
            </widget>
            <widget>
                <identifier>123</identifier>
                <visibility>hidden</visibility>
            </widget>
            <widget>
                <identifier>456</identifier>
                <visibility>vacant</visibility>
            </widget>
        </homePage>
        <objectForms>
            <objectForm>
                <type>OrgType</type>
                <formSpecification>
                    <title>Transform</title>
                    <panelClass>com.example.midpoint.gui.forms.ConfigurationTableTabPanel</panelClass>
                </formSpecification>
            </objectForm>
        </objectForms>
        <objectCollectionViews>
            <objectCollectionView id="3">
                <action id="4">
                    <identifier>action1</identifier>
                </action>
                <action id="5">
                    <identifier>action2</identifier>
                </action>
                <searchBoxConfiguration>
                    <scopeConfiguration>
                        <defaultValue>oneLevel</defaultValue>
                    </scopeConfiguration>
                    <objectTypeConfiguration>
                        <defaultValue>UserType</defaultValue>
                    </objectTypeConfiguration>
                    <searchItems>
                        <searchItem id="7">
                            <display>
                                <label>sample1</label>
                            </display>
                        </searchItem>
                    </searchItems>
                </searchBoxConfiguration>
            </objectCollectionView>
            <objectCollectionView id="6">
                <searchBoxConfiguration>
                    <scopeConfiguration>
                        <defaultValue>subtree</defaultValue>
                    </scopeConfiguration>
                    <objectTypeConfiguration>
                        <defaultValue>OrgType</defaultValue>
                    </objectTypeConfiguration>
                    <searchItems>
                        <searchItem id="8">
                            <display>
                                <label>sample3</label>
                            </display>
                        </searchItem>
                    </searchItems>
                </searchBoxConfiguration>
            </objectCollectionView>
            <objectCollectionView id="15">
                <identifier>orgMember</identifier>
                <type>OrgType</type>
            </objectCollectionView>
        </objectCollectionViews>
        <objectDetails>
            <objectDetailsPage id="10">
                <type>UserType</type>
                <forms id="16">
                    <type>UserType</type>
                    <formSpecification>
                        <title>Transform</title>
                        <panelClass>com.example.midpoint.gui.forms.ConfigurationTableTabPanel</panelClass>
                    </formSpecification>
                </forms>
                <container id="9">
                    <identifier>123</identifier>
                </container>
            </objectDetailsPage>
            <objectDetailsPage>
                <type>OrgType</type>
                <panel>
                    <identifier>orgMembers</identifier>
                    <listView>
                        <searchBoxConfiguration>
                            <scopeConfiguration>
                                <display>
                                    <label>Custom Scope</label>
                                    <help>Help scope text</help>
                                </display>
                                <defaultValue>subtree</defaultValue>
                            </scopeConfiguration>
                            <objectTypeConfiguration>
                                <display>
                                    <label>Custom Type</label>
                                </display>
                                <defaultValue>OrgType</defaultValue>
                                <supportedTypes>OrgType</supportedTypes>
                                <supportedTypes>UserType</supportedTypes>
                                <supportedTypes>ServiceType</supportedTypes>
                            </objectTypeConfiguration>
                            <indirectConfiguration>
                                <visibility>hidden</visibility>
                            </indirectConfiguration>
                        </searchBoxConfiguration>
                    </listView>
                </panel>
            </objectDetailsPage>
        </objectDetails>
        <accessRequest>
            <roleCatalog>
                <roleCatalogRef oid="0e5b7304-ea5c-438e-84d1-2b0ce40517ce" type="OrgType"/>
                <collection>
                    <identifier>allRoles</identifier>
                    <default>true</default>
                    <collectionIdentifier>allRoles</collectionIdentifier>
                </collection>
                <collection>
                    <identifier>myCustomCollection</identifier>
                    <collectionIdentifier>myCustomCollection</collectionIdentifier>
                </collection>
            </roleCatalog>
        </accessRequest>
    </adminGuiConfiguration>

    <defaultObjectPolicyConfiguration id="2">
        <itemConstraint>
            <path>name</path>
            <visibility>visible</visibility>
        </itemConstraint>
        <itemConstraint>
            <path>givenName</path>
            <visibility>hidden</visibility>
        </itemConstraint>
        <itemConstraint>
            <path>familyName</path>
            <visibility>visible</visibility>
        </itemConstraint>
    </defaultObjectPolicyConfiguration>

    <cleanupPolicy>
        <simpleOperationExecutions>
            <maxAge>P1D</maxAge>
            <maxRecords>10</maxRecords>
        </simpleOperationExecutions>
        <complexOperationExecutions>
            <maxAge>P1D</maxAge>
            <maxRecords>10</maxRecords>
        </complexOperationExecutions>
    </cleanupPolicy>
</systemConfiguration>
